airflow/api_fastapi/core_api/datamodels/connections.py

```diff
@@ -90,8 +91,74 @@ class ConnectionBody(BaseModel):
     extra: str | None = Field(default=None)
 
 
-class ConnectionBulkBody(BaseModel):
-    """Connections Serializer for requests body."""
+class ConnectionBulkCreateAction(BaseModel):
+    """Bulk Create Variable serializer for request bodies."""
+
+    action: Literal["create"] = "create"
+    connections: list[ConnectionBody] = Field(..., description="A list of connections to be created.")
+    action_if_exists: Literal["skip", "overwrite", "fail"] = "fail"
+
+
+class ConnectionBulkUpdateAction(BaseModel):
+    """Bulk Update Connection serializer for request bodies."""
+
+    action: Literal["update"] = "update"
+    connections: list[ConnectionBody] = Field(..., description="A list of connections to be updated.")
+    action_if_not_exists: Literal["skip", "fail"] = "fail"
 
-    connections: list[ConnectionBody]
-    overwrite: bool | None = Field(default=False)
+
+class ConnectionBulkDeleteAction(BaseModel):
+    """Bulk Delete Connection serializer for request bodies."""
+
+    action: Literal["delete"] = "delete"
+    connection_ids: list[str] = Field(..., description="A list of connection IDs to be deleted.")
+    action_if_not_exists: Literal["skip", "fail"] = "fail"
+
+
+class ConnectionBulkBody(BaseModel):
+    """Request body for bulk Connection operations (create, update, delete)."""
+
+    actions: list[ConnectionBulkCreateAction | ConnectionBulkUpdateAction | ConnectionBulkDeleteAction] = (
```

Comment:
I think we can add a pydantic discriminator to save time at the serialization time. Basically the `action` field tells us what model should be use to deserialize an action.

Reply:
https://docs.pydantic.dev/latest/concepts/unions/